### ISSUE TITLE
Provide a step by step upgrade guide for Ember Data 1.13

### ISSUE THREAD

**Issue by Gaurav0** [emberjs/website#2355]
The blog post here: http://emberjs.com/blog/2015/06/18/ember-data-1-13-released.html promises a detailed, step by step upgrade guide to Ember Data 1.13. Despite the fact that Ember Data 2.0 is out, this has not been published.

What is most desperately needed:

* An explanation of the JSON-API data structure's differences from REST
* A more detailed guide to upgrading to new serializer api, this is *really* the most difficult part for me
* An update to the blog post containing changes since 1.13 (in 1.13.x, mostly bugfixes, but one significant rollback)
* An explanation on how to deal with converting a large application, i.e. how to change over the serializer API for one model at a time
* A new guide for custom extending of JSONAPIAdapter and JSONAPISerializer. This is one area where I think the blog post particularly falls short.
* A chart of renamed methods by class would be helpful
* Where should one start? What parts should be left till later? (Hint: serializer api should be after other stuff)
* I raised this elsewhere, but this ties into a need for a rewrite of guides model section

**Comment by wecc** [comment]
> A more detailed guide to upgrading to new serializer api, this is really the most difficult part for me

Following the steps outlined in the blog post, what was the biggest pain when upgrading to the new Serializer API?

**Comment by Gaurav0** [comment]
Things are really tied together here. For me, I am just so unfamiliar with the new JSON API format, and all I get is a link to the huge json api doc. Based on the blog post, I renamed some methods, set the new serializer flag, and then wondered what to do after everything broke with basic js errors like "undefined is not a function", so I just reverted for now.

My guess is my returned payloads aren't in the expected format. Maybe a "if you have this it should now look like this" kind of section?

The other problem is that I have very little understanding of writing custom adapters and serializers in Ember Data the first place (which has never been made into a good tutorial by anyone, I think Igor's one at EmberConf is the only one I ever found and it cost $350), and I'm trying to upgrade other people's code.

Unfortunately since I never really solved my issues, and mostly venting, I don't know best how to go about telling you how to write the guide. But I hope this helps.
